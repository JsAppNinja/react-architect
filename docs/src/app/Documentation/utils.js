export const splitOnCaps = s => s.split(/(?=[A-Z1-9])/);

function getString(s) {
  return !(s instanceof Array) ? splitOnCaps(s) : s;
}

export const toDashedName = (s) => getString(s).map(s => s.toLowerCase()).join('-');
export const toTitle = (s) => getString(s).join(' ');
