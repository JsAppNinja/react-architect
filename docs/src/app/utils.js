import React from 'react';
export const githubHref = 'https://github.com/mlaursen/react-md';

const lorems = [
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at placerat nulla. Morbi eu erat diam. Donec malesuada mauris in neque efficitur luctus. Vestibulum porttitor elit velit, pellentesque imperdiet tellus maximus in. Phasellus consectetur lacus nunc, quis efficitur felis feugiat ac. Vivamus non erat laoreet, volutpat tellus sit amet, porttitor orci. Sed scelerisque id arcu at ornare. Suspendisse potenti. Phasellus eu posuere mi. Fusce venenatis odio dui, ac accumsan lacus dapibus et. Praesent euismod nulla at porta elementum. Nunc ullamcorper velit aliquam velit faucibus tempus.',
  'Duis ligula elit, elementum a nisl a, venenatis fermentum libero. Cras bibendum dignissim nisl, sed sodales tellus scelerisque sed. Sed at enim semper, imperdiet elit et, venenatis magna. Morbi vestibulum purus in ante placerat viverra. Nullam molestie arcu ut dolor rutrum varius. Sed vehicula laoreet volutpat. Quisque feugiat, nisl in sollicitudin consectetur, mi metus aliquet urna, ac sagittis nulla nulla in diam. Fusce id enim auctor ante fringilla auctor in et quam.',
  'Donec pulvinar ligula augue, et tincidunt velit bibendum non. Aenean id metus nunc. Donec non feugiat nisl. In accumsan ligula purus, ut vulputate elit tempus sit amet. Nullam volutpat semper lectus. Nunc a libero tellus. Duis posuere nisi diam, nec euismod risus interdum sit amet. Pellentesque at lacus bibendum, cursus ante non, maximus arcu. Morbi id leo nulla. Integer quis feugiat quam. Etiam auctor pharetra vestibulum.',
  'Nulla sodales metus metus. Sed a metus quis ipsum sagittis placerat. Nunc sit amet molestie purus, nec rutrum quam. Praesent elementum risus eget mattis condimentum. Morbi dignissim, dolor et finibus sollicitudin, nulla enim molestie est, eget tincidunt magna libero et dolor. Fusce tempor sapien et urna sodales, at laoreet nunc egestas. Aenean eget arcu in felis ultrices vehicula. Maecenas interdum viverra sem vitae sodales. Duis tincidunt varius nunc sed viverra. Duis sit amet semper ex, ultrices aliquet mi. Donec a congue sem, ac facilisis ex.',
  'Proin venenatis tortor et ex viverra, ac facilisis mi ullamcorper. Aenean eget est bibendum, fermentum enim eget, semper orci. Nunc magna mauris, fermentum sit amet diam ut, semper malesuada justo. Fusce faucibus tellus eget arcu imperdiet, aliquet placerat massa molestie. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Maecenas gravida quam id ultricies accumsan. Sed porta accumsan libero sed venenatis. Morbi scelerisque, ex in gravida porttitor, ipsum erat sodales urna, mattis mattis nibh ligula quis tellus. Pellentesque consequat et metus scelerisque dapibus. Praesent at maximus ligula. Phasellus vel orci tortor.',
  'Mauris posuere nisl sit amet justo condimentum volutpat. Aliquam venenatis rhoncus est, tristique semper tortor iaculis et. Donec pretium metus nec sodales mollis. Ut vel ipsum sapien. Fusce sit amet libero efficitur orci aliquet scelerisque a ut arcu. In hac habitasse platea dictumst. Sed turpis libero, dictum nec imperdiet nec, maximus eget urna. Sed vel lorem sit amet nisi vulputate mattis. In in odio erat. Aenean euismod lorem nec viverra auctor. Mauris id ipsum maximus, tempus dui nec, mollis felis. Etiam sodales metus sit amet lacus aliquet tristique. Nulla in lacus massa. Morbi quis lorem vitae dolor maximus luctus vel vel eros. Phasellus posuere in purus eget convallis.',
  'Aliquam mauris justo, ornare quis quam fermentum, mollis volutpat dolor. Donec sagittis velit non nisl porttitor, id lacinia tortor vulputate. Suspendisse posuere at felis eu fermentum. Donec at vehicula leo, id sollicitudin ipsum. Donec dapibus sodales erat at malesuada. Mauris sollicitudin velit sed arcu molestie congue. Suspendisse ac arcu enim. Sed ac urna id mi rutrum ornare sit amet in nisl. Phasellus at tempus magna. Nulla facilisi. Mauris pharetra finibus mi ut facilisis. Duis mattis odio id quam vestibulum, id consequat ex fringilla. Aenean sollicitudin accumsan sollicitudin. Duis euismod maximus tempor. Phasellus mi enim, bibendum at nisi vel, cursus venenatis sem. Praesent sagittis, nunc at rhoncus maximus, ligula turpis imperdiet eros, pharetra tincidunt tortor magna id sem.',
  'Sed eleifend sapien condimentum iaculis cursus. In vel pretium turpis. Duis molestie euismod turpis id consectetur. Nunc pharetra sapien libero, a volutpat erat finibus eget. Duis euismod eleifend felis consequat fringilla. Sed varius diam ligula, quis rutrum ex mollis non. Phasellus ac malesuada nibh. Praesent sit amet auctor urna. Quisque eget magna id ex dapibus vulputate eget vel magna.',
  'Ut viverra tempor placerat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras neque diam, auctor eget sapien et, tincidunt blandit turpis. Nam non volutpat velit. Aliquam iaculis euismod nisl non facilisis. Ut blandit dignissim turpis in finibus. Cras neque augue, tempus non neque eu, ornare ultrices sem. Nullam ultricies mi quis semper pellentesque. Mauris mauris lacus, ullamcorper a iaculis quis, tempus id tellus. Phasellus in orci orci. Fusce consequat suscipit ipsum ac bibendum. Maecenas et leo eu leo iaculis varius. Nam consectetur laoreet nisl blandit condimentum.',
  'Ut dictum, sapien vel malesuada posuere, tortor tellus blandit risus, eget hendrerit quam diam eu nibh. Aliquam erat volutpat. Ut rhoncus augue ac nunc porttitor hendrerit. Pellentesque enim risus, auctor quis turpis eget, pulvinar malesuada urna. Nullam tortor mi, elementum id sodales eget, aliquam et nulla. Pellentesque efficitur mi eros, in rutrum metus ultrices vel. Proin vitae diam sit amet nunc pretium malesuada. Fusce at nisi scelerisque, ultrices tortor nec, finibus velit. Donec molestie augue eget leo rutrum, nec ornare arcu gravida. Quisque tincidunt risus elit, ut condimentum eros egestas congue.',
];
export function loremIpsum(paras = 3) {
  let ipsums = [];
  for(let i = 0; i < paras; i++) {
    ipsums.push((
      <p key={i}>{lorems[Math.floor(Math.random() * lorems.length)]}</p>
    ));
  }
  return ipsums;
}

export const numstr = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'];

export const randomImage = ({ width, height, time } = {}) => {
  width = !width ? 200 : width;
  height = !height ? 200 : height;
  time = typeof time === 'undefined' ? Date.now() : time;
  return `https://unsplash.it/${width}/${height}/?random&time=${time}`;
};

export const hostPrefix = ''; // change to '/react-md' for production
export const imgPrefix = hostPrefix + '/imgs';
