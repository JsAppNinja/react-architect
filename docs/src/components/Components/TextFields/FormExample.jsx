import React, { PureComponent } from 'react';
import Button from 'react-md/lib/Buttons/Button';
import CardActions from 'react-md/lib/Cards/CardActions';
import SelectionControl from 'react-md/lib/SelectionControls/SelectionControl';
import SelectionControlGroup from 'react-md/lib/SelectionControls/SelectionControlGroup';
import Snackbar from 'react-md/lib/Snackbars';
import TextField from 'react-md/lib/TextFields';

import PhoneEmulator from 'components/PhoneEmulator';

import './_form-styles.scss';

const controls = [{
  label: 'My amazing store',
  value: 'MAS',
}, {
  label: 'Amazon',
  value: 'AMZ',
}, {
  label: 'A pretty terrible store',
  value: 'PTS',
}, {
  label: 'EBay',
  value: 'EBY',
}, {
  label: 'Some other store',
  value: 'SOS',
}];

export default class FormExample extends PureComponent {
  state = { toasts: [] };

  handleSubmit = (e) => {
    e.preventDefault();
    if (!this.state.toasts.length) {
      const toasts = this.state.toasts.slice();
      toasts.push({
        text: 'Submitted new application',
        action: 'Neat!',
      });
      this.setState({ toasts });
    }
  };

  handleDismiss = () => {
    const [, ...toasts] = this.state.toasts;
    this.setState({ toasts });
  };

  render() {
    const { toasts } = this.state;

    return (
      <PhoneEmulator toolbarTitle="Application">
        <form className="md-grid text-fields__application" onSubmit={this.handleSubmit}>
          <TextField
            id="application-title"
            label="Title"
            defaultValue="Vintage 50's Dress"
            customSize="title"
            className="md-cell md-cell--12"
            required
          />
          <TextField
            id="application-price"
            label="Price"
            type="number"
            defaultValue={10}
            step={0.01}
            min={0}
            pattern="^d+(\.|\,)\d{2}"
            className="text-fields__application__price md-cell md-cell--3 md-cell--1-phone"
            required
          />
          <TextField
            id="application-location"
            label="Location (optional)"
            defaultValue="Fremont Bridge"
            className="md-cell md-cell--9 md-cell--3-phone md-cell--5-tablet"
          />
          <TextField
            id="application-description"
            label="Description"
            rows={2}
            maxRows={6}
            defaultValue="Unique and rare dress from 1952. Made out of cotton with front pockets. Sleeveless with button closures."
            className="md-cell md-cell--12"
          />
          <SelectionControl
            id="application-publish"
            type="switch"
            label="Publish immediately?"
            name="publishing"
            inline
            className="md-cell md-cell--12"
          />
          <SelectionControlGroup
            label="Publish to:"
            id="publish-to"
            name="publish-places"
            controls={controls}
            type="checkbox"
            className="md-cell md-cell--12 text-fields__application__publishes"
            defaultValue="MAS,EBY"
          />
          <CardActions className="md-cell md-cell--12">
            <Button raised primary type="submit" className="md-cell--right">Submit</Button>
          </CardActions>
        </form>
        <Snackbar id="application-toasts" toasts={toasts} onDismiss={this.handleDismiss} />
      </PhoneEmulator>
    );
  }
}
