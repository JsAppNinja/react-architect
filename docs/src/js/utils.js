import React from 'react';

import { FontIcon, Avatar } from 'react-md';

export const githubHref = 'https://github.com/mlaursen/react-md';

export const mainLinks = [{
  link: '',
  primaryText: 'Home',
  leftIcon: <FontIcon>home</FontIcon>,
}, {
  link: 'sassdoc',
  primaryText: 'SASS Doc',
  leftAvatar: <Avatar src="/imgs/sass-icon.png" alt="SASS Icon" />,
}];
