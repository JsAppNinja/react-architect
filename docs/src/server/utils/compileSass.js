const path = require('path');
const Promise = require('bluebird');
const nodeSass = require('node-sass');

const sass = Promise.promisify(nodeSass.render);

const { ACCENTABLE_COLORS, PRIMARY_COLORS, SECONDARY_HUES } = require('../../shared/constants/colors');

const DARK_THEME_STYLES = `
.example-code,
.code-block {
  background: $md-grey-300;
}

a:not(.md-btn) {
  color: $md-purple-a-400;

  &:visited {
    color: $md-purple-a-100;
  }
}
`;

function toColorName(color) {
  return color && color.replace('_', '-');
}

function validatePrimary(color) {
  return color && PRIMARY_COLORS.indexOf(color) !== -1;
}

function validateSecondary(color, hue) {
  return color && hue && ACCENTABLE_COLORS.indexOf(color) !== -1 && SECONDARY_HUES.indexOf(hue) !== -1;
}

/**
 * Attempts to compiles sass based off of a css file name. The filename should be formatted
 * as 'PRIMARY_COLOR-SECONDARY_COLOR-HUE-dark?.css' where `dark` is optional.
 *
 * @param {String} name - The CSS filename to convert into styles.
 * @return {Promise} a promise with the buffer of compiles styles
 */
module.exports = function compileSass(name) {
  const [p, s, hue, dark] = name.split('-');
  const primary = toColorName(p);
  const secondary = toColorName(s);

  if (!validatePrimary(primary)) {
    return Promise.reject(new Error(`Invalid primary color: ${primary}`));
  } else if (!validateSecondary(secondary, parseInt(hue, 10))) {
    return Promise.reject(new Error(`Invalid secondary color: ${secondary}-${hue}`));
  }

  const primaryColor = `$md-${primary}-500`;
  const secondaryColor = `$md-${secondary}-a-${hue}`;
  const darkThemeStyles = dark ? DARK_THEME_STYLES : '';

  const styles = `
@import 'react-md';

$md-primary-color: $md-light-blue-500;
$md-secondary-color: $md-deep-orange-a-200;
@include react-md-theme-everything(${primaryColor}, ${secondaryColor}, ${!dark}, 'custom-theme');

.custom-theme {
  .banner {
    background: ${primaryColor};
  }

  .react-md-logo {
    text,
    #Group > path:first-child,
    path#Oval-210 {
      fill: ${secondaryColor};
    }
  }

  ${darkThemeStyles}
}

@media #{$md-tablet-media} {
  .custom-theme .phone-size-container .md-toolbar:not(.md-toolbar--inset)::before {
    background: ${primaryColor};
  }
}
  `;

  return sass({
    data: styles,
    includePaths: [path.resolve(process.cwd(), '..', 'src', 'scss')],
    outputStyle: 'compressed',
  }).then((stats, error) => {
    if (error) {
      return Promise.reject(error);
    }

    return stats.css;
  });
};
