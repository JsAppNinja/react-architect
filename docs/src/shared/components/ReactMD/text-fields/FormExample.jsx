import React from 'react';
import TextField from 'react-md/lib/TextFields';
import Button from 'react-md/lib/Buttons/Button';

import PhoneSizeDemo from 'containers/PhoneSizeDemo';
import './_application.scss';

const FormExample = () => (
  <PhoneSizeDemo
    title="Application"
    iconLeft="arrow_back"
    toolbarActions={<Button icon>more_vert</Button>}
  >
    <form className="md-grid">
      <TextField
        id="applicationTitle"
        label="Title"
        defaultValue="Vintage 50's Dress"
        customSize="title"
        className="md-cell md-cell--12"
        required
      />
      <TextField
        id="applicationPrice"
        label="Price"
        type="number"
        defaultValue={10}
        step={0.01}
        min={0}
        pattern="^\d+(\.|\,)\d{2}"
        className="price md-cell md-cell--1-phone md-cell--3"
      />
      <TextField
        id="applicationLocation"
        label="Location (optional)"
        defaultValue="Fremont Bridge"
        className="location md-cell md-cell--3-phone md-cell--5-tablet md-cell--9-desktop"
      />
      <TextField
        id="applicationDescription"
        label="Description"
        rows={2}
        defaultValue="Unique and rare dress from 1952. Made out of cotton with front pockets. Sleeveless with button closures."
        className="md-cell md-cell--12"
      />
    </form>
  </PhoneSizeDemo>
);

export default FormExample;
