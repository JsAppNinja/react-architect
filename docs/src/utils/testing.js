/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { StaticRouter } from 'react-router';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { render, shallow, mount } from 'enzyme';
import rootReducer from 'state';

/**
 * I'm lazy.
 *
 * Could be lazier and do `expectSnapshot` though..
 */
export function createSnapshot(children) {
  return renderer.create(children).toJSON();
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * uses something from react-router.
 */
export function createRouterSnapshot(children, location = '/', context = {}) {
  return createSnapshot(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * requires the redux provider.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function createReduxSnapshot(children, state) {
  const store = createStore(rootReducer, state);
  return createSnapshot(
    <Provider store={store}>
      {children}
    </Provider>
  );
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * requires both the redux Provider and something from react-router.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function createReduxRouterSnapshot(children, state, location = '/', context = {}) {
  const store = createStore(rootReducer, state);
  return createSnapshot(
    <Provider store={store}>
      <StaticRouter location={location} context={context}>
        {children}
      </StaticRouter>
    </Provider>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component relies on react-router. This should be used when a component
 * or child component uses findDOMNode or the CSSTransitionGroup, otherwise the createRouterSnapshot should
 * be used.
 */
export function renderRouterSnapshot(children, location = '/', context = {}) {
  return render(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component requires the redux Provider. This should be used when a component
 * or child component uses findDOMNode or the CSSTransitionGroup, otherwise the createReduxSnapshot should
 * be used.
 */
export function renderReduxSnapshot(children, state) {
  const store = createStore(rootReducer, state);
  return render(
    <Provider store={store}>
      {children}
    </Provider>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component requires both the redux Provider and something from react-router.
 * This should be used when a component * or child component uses findDOMNode or the CSSTransitionGroup,
 * otherwise the createReduxRouterSnapshot should be used.
 */
export function renderReduxRouterSnapshot(children, state, location = '/', context = {}) {
  const store = createStore(rootReducer, state);
  return render(
    <Provider store={store}>
      <StaticRouter location={location} context={context}>
        {children}
      </StaticRouter>
    </Provider>
  );
}

/**
 * A simple wrapper to use enzyme's shallow rendering when the component or a child
 * requires something from react-router.
 */
export function shallowWithRouter(children, location = '/', context = {}) {
  return shallow(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}


/**
 * A simple wrapper to use enzyme's mount rendering when the component or a child
 * requires something from react-router.
 */
export function mountWithRouter(children, location = '/', context = {}) {
  return mount(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * A simple wrapper to use enzyme's shallow rendering when the component or a child
 * requires something from react-redux.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function shallowWithProvider(children, state) {
  const store = createStore(rootReducer, state);
  return shallow(<Provider store={store}>{children}</Provider>);
}

/**
 * A simple wrapper to use enzyme's mount rendering when the component or a child
 * requires something from react-redux.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function mountWithProvider(children, state) {
  const store = createStore(rootReducer, state);
  return mount(<Provider store={store}>{children}</Provider>);
}

/**
 * A utility function that will override the console's behavior and throw errors
 * if console.error occurs. This is useful when you want to consider React warnings
 * as failures.
 */
export function captureConsole(match = null) {
  const console = global.console;

  beforeAll(() => {
    global.console = {
      ...console,
      error: (error) => {
        if (match === null || error.match(match)) {
          throw new Error(error);
        }
      },
    };
  });

  afterAll(() => {
    global.console = console;
  });
}
