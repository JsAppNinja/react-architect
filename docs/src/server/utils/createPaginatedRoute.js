import { BAD_REQUEST } from 'constants/responseCodes';
import { getUrl } from 'server/utils/getServerUrl';
import { omit } from 'lodash';

function isInt(x) {
  return !Number.isNaN(x) && x % 1 === 0;
}

/**
 * Creates an express route handler for paginating data. The paginated data response will return
 * an object containing meta information and the list of data.
 *
 * The metadata will contain the requested start index, the requested limit, the total number of results,
 * an optional next link, and an option previous link. This will allow the client to automatically request
 * the next dataset when needed.
 *
 * @param {function} getData - a function to get the data for the current route. It will provide the current
 *    query string.
 * @param {Object=} config - Any additional config to use.
 * @param {boolean=true} config.queryable - boolean if the route should be queryable.
 * @param {number=50} config.maxResults - the maximun number of results allowed.
 * @param {Array.<String>|String=} config.omitKeys - An optional list or sintr string of keys to remove
 *    from the response.
 * @return {function} a route handler for getting paginated data.
 */
export default function createPaginatedRoute(getData, { queryable = true, maxResults = 50, omitKeys = null } = {}) {
  return function handlePaginationRequest(req, res) {
    const { q } = req.query;
    let { start, limit } = req.query;

    if (!start) {
      start = 0;
    }

    if (!limit) {
      limit = 10;
    }

    if ((queryable && !q) || !isInt(start) || !isInt(limit) || start < 0 || limit > maxResults || limit < 1) {
      res.sendStatus(BAD_REQUEST);
      return;
    }

    start = parseInt(start, 10);
    limit = parseInt(limit, 10);

    const data = getData(q);
    const total = data.length;
    if (start > total) {
      res.sendStatus(BAD_REQUEST);
      return;
    }

    let url = `${getUrl(req).replace(/\?.*/, '')}?`;
    if (queryable) {
      url = `${url}q=${q}&`;
    }

    let next = null;
    let previous = null;
    if (total > start + limit) {
      next = `${url}start=${start + limit}&limit=${limit}`;
    }

    if (start + limit > limit) {
      previous = `${url}start=${Math.max(0, start - limit)}&limit=${limit}`;
    }

    const sliced = data.slice(start, start + limit);

    res.json({
      meta: {
        start,
        limit,
        total,
        next,
        previous,
      },
      data: omitKeys && omitKeys.length ? sliced.map(datum => omit(datum, omitKeys)) : sliced,
    });
  };
}
