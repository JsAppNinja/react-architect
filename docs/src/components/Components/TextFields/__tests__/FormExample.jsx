/* eslint-env jest */
import React from 'react';
import { renderReduxSnapshot } from 'utils/testing';

import FormExample from '../FormExample';

describe('FormExample', () => {
  it('should render correctly', () => {
    const tree = renderReduxSnapshot(<FormExample />);
    expect(tree).toMatchSnapshot();
  });
});
