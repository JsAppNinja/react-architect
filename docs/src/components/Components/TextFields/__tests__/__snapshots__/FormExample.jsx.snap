// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormExample should render correctly 1`] = `
<div
  class="md-paper md-paper--1 phone-emulator md-background"
>
  <header
    class="phone-emulator__status-bar"
  >
    <i
      class="md-icon material-icons"
    >
      network_wifi
    </i>
    <i
      class="md-icon material-icons"
    >
      network_cell
    </i>
    <i
      class="md-icon material-icons"
    >
      battery_full
    </i>
    <span
      class="phone-emulator__status-bar__time"
    >
      12:30
    </span>
  </header>
  <header
    class="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--fixed phone-emulator__toolbar"
  >
    <button
      class="md-btn md-btn--icon md-pointer--hover md-inline-block md-btn--toolbar md-toolbar--action-left"
      type="button"
    >
      <div
        class="md-ink-container"
      />
      <i
        class="md-icon material-icons md-text--inherit"
      >
        menu
      </i>
    </button>
    <h2
      class="md-title md-title--toolbar"
    >
      Application
    </h2>
  </header>
  <section
    class="phone-emulator__content phone-emulator__content--offset"
  >
    <form
      class="md-grid text-fields__application"
    >
      <div
        class="md-text-field-container md-full-width md-text-field-container--input md-cell md-cell--12"
      >
        <label
          class="md-floating-label md-floating-label--title md-floating-label--floating md-text--secondary"
          for="application-title"
        >
          Title *
        </label>
        <input
          class="md-text-field md-text-field--floating-margin md-text-field--title md-full-width md-text"
          id="application-title"
          type="text"
          value="Vintage 50's Dress"
        />
        <hr
          class="md-divider md-divider--text-field md-divider--expand-from-left"
        />
      </div>
      <div
        class="md-text-field-container md-full-width md-text-field-container--input text-fields__application__price md-cell md-cell--3 md-cell--1-phone"
      >
        <label
          class="md-floating-label md-floating-label--floating md-text--secondary"
          for="application-price"
        >
          Price *
        </label>
        <input
          class="md-text-field md-text-field--floating-margin md-full-width md-text"
          id="application-price"
          min="0"
          pattern="^d+(\\\\.|\\\\,)\\\\d{2}"
          step="0.01"
          type="number"
          value="10"
        />
        <hr
          class="md-divider md-divider--text-field md-divider--expand-from-left"
        />
      </div>
      <div
        class="md-text-field-container md-full-width md-text-field-container--input md-cell md-cell--9 md-cell--3-phone md-cell--5-tablet"
      >
        <label
          class="md-floating-label md-floating-label--floating md-text--secondary"
          for="application-location"
        >
          Location (optional)
        </label>
        <input
          class="md-text-field md-text-field--floating-margin md-full-width md-text"
          id="application-location"
          type="text"
          value="Fremont Bridge"
        />
        <hr
          class="md-divider md-divider--text-field md-divider--expand-from-left"
        />
      </div>
      <div
        class="md-text-field-container md-full-width md-text-field-container--multiline md-cell md-cell--12"
      >
        <label
          class="md-floating-label md-floating-label--floating md-text--secondary"
          for="application-description"
        >
          Description
        </label>
        <div
          class="md-text-field-multiline-container md-text-field--floating-margin"
        >
          <span
            aria-hidden="true"
          />
          <textarea
            aria-hidden="true"
            class="md-text-field md-text-field--multiline md-full-width md-text md-text-field--multiline-mask"
            readonly=""
            rows="2"
            tabindex="-1"
          >
            Unique and rare dress from 1952. Made out of cotton with front pockets. Sleeveless with button closures.
          </textarea>
          <textarea
            class="md-text-field md-text-field--multiline md-full-width md-text"
            id="application-description"
            rows="2"
          >
            Unique and rare dress from 1952. Made out of cotton with front pockets. Sleeveless with button closures.
          </textarea>
        </div>
        <hr
          class="md-divider md-divider--text-field md-divider--expand-from-left"
        />
      </div>
      <div
        class="md-selection-control-container md-selection-control-container--inline md-switch-container md-cell md-cell--12"
      >
        <input
          aria-hidden="true"
          class="md-selection-control-input"
          id="application-publish"
          name="publishing"
          type="checkbox"
        />
        <label
          class="md-selection-control-label md-pointer--hover md-text"
          for="application-publish"
        >
          <div
            class="md-switch-track md-pointer--hover md-switch-track--off"
          >
            <div
              aria-pressed="false"
              class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-switch-thumb md-switch-thumb--off"
              role="button"
              tabindex="0"
            >
              <div
                class="md-ink-container md-ink-container--2x"
              />
            </div>
          </div>
          <span>
            Publish immediately?
          </span>
        </label>
      </div>
      <fieldset
        class="md-selection-control-group md-cell md-cell--12 text-fields__application__publishes"
      >
        <legend
          class="md-subheading-1"
        >
          Publish to:
        </legend>
        <div
          class="md-selection-control-container"
        >
          <input
            aria-hidden="true"
            checked=""
            class="md-selection-control-input"
            id="publish-to0"
            name="publish-places[]"
            type="checkbox"
            value="MAS"
          />
          <label
            class="md-selection-control-label md-pointer--hover md-text"
            for="publish-to0"
          >
            <div
              aria-checked="true"
              aria-pressed="false"
              class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="md-ink-container"
              />
              <i
                class="md-icon material-icons md-text--inherit"
              >
                check_box
              </i>
            </div>
            <span>
              My amazing store
            </span>
          </label>
        </div>
        <div
          class="md-selection-control-container"
        >
          <input
            aria-hidden="true"
            class="md-selection-control-input"
            id="publish-to1"
            name="publish-places[]"
            type="checkbox"
            value="AMZ"
          />
          <label
            class="md-selection-control-label md-pointer--hover md-text"
            for="publish-to1"
          >
            <div
              aria-checked="false"
              aria-pressed="false"
              class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="md-ink-container"
              />
              <i
                class="md-icon material-icons md-text--inherit"
              >
                check_box_outline_blank
              </i>
            </div>
            <span>
              Amazon
            </span>
          </label>
        </div>
        <div
          class="md-selection-control-container"
        >
          <input
            aria-hidden="true"
            class="md-selection-control-input"
            id="publish-to2"
            name="publish-places[]"
            type="checkbox"
            value="PTS"
          />
          <label
            class="md-selection-control-label md-pointer--hover md-text"
            for="publish-to2"
          >
            <div
              aria-checked="false"
              aria-pressed="false"
              class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="md-ink-container"
              />
              <i
                class="md-icon material-icons md-text--inherit"
              >
                check_box_outline_blank
              </i>
            </div>
            <span>
              A pretty terrible store
            </span>
          </label>
        </div>
        <div
          class="md-selection-control-container"
        >
          <input
            aria-hidden="true"
            checked=""
            class="md-selection-control-input"
            id="publish-to3"
            name="publish-places[]"
            type="checkbox"
            value="EBY"
          />
          <label
            class="md-selection-control-label md-pointer--hover md-text"
            for="publish-to3"
          >
            <div
              aria-checked="true"
              aria-pressed="false"
              class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="md-ink-container"
              />
              <i
                class="md-icon material-icons md-text--inherit"
              >
                check_box
              </i>
            </div>
            <span>
              EBay
            </span>
          </label>
        </div>
        <div
          class="md-selection-control-container"
        >
          <input
            aria-hidden="true"
            class="md-selection-control-input"
            id="publish-to4"
            name="publish-places[]"
            type="checkbox"
            value="SOS"
          />
          <label
            class="md-selection-control-label md-pointer--hover md-text"
            for="publish-to4"
          >
            <div
              aria-checked="false"
              aria-pressed="false"
              class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary"
              role="checkbox"
              tabindex="0"
            >
              <div
                class="md-ink-container"
              />
              <i
                class="md-icon material-icons md-text--inherit"
              >
                check_box_outline_blank
              </i>
            </div>
            <span>
              Some other store
            </span>
          </label>
        </div>
      </fieldset>
      <section
        class="md-dialog-footer--card md-dialog-footer--inline md-cell md-cell--12"
      >
        <button
          class="md-btn md-btn--raised md-btn--text md-pointer--hover md-background--primary md-background--primary-hover md-inline-block md-cell--right"
          type="submit"
        >
          <div
            class="md-ink-container"
          />
          Submit
        </button>
      </section>
    </form>
  </section>
</div>
`;
